src/plugins/intel_npu/tools/common/include/tools_helpers.hpp

```diff
@@ -74,74 +74,95 @@ std::map<std::string, std::vector<std::string>> parseInputParameters(std::string
+
+    // shape and override_model_batch_size cannot be specificed together
+    if (!shapeString.empty() && overrideModelBatchSize != 1) {
+        throw std::logic_error(R"(Incompatible params: "shape" and "override_model_batch_size")");
     }
-    std::cout << "Configuring model batch: " << batch << std::endl;
-    for (auto&& item : model->get_parameters()) {
-        auto shape = item->get_partial_shape();
-        auto rank = shape.rank();
-        if (rank.is_dynamic()) {
-            throw std::logic_error("Rank \"" + rank.to_string() + "\" of the shape \"" + shape.to_string() +
-                                   "\" is dynamic which is not supported by NPU");
-        }
-        auto layout = item->get_layout();
-        if (!ov::layout::has_batch(layout)) {
-            item->set_layout(ov::Layout(layout.to_string().insert(1, "N,")));
+
+    // If override_model_batch_size is specified (default is 1):
+    // 1. Get the layout of the model's parameters and check for batch dimension
+    // 2. If batch dimension is not found, insert 'N' dimension = 1 to the layout
+    // 3. Set the shape at batch index to match overrideModelBatchSize
+    // 4. Pass the shape string as `shapeString` to be used as if user specified shape
+    if (overrideModelBatchSize != 1) {
+        for (auto&& item : model->get_parameters()) {
+            auto layout = item->get_layout();
+            if (!ov::layout::has_batch(layout)) {
+                std::cout << "WARNING: Batch layout not found. Inserting 'N' dimension = 1 to the layout before "
+                          << "setting new model batch of " << overrideModelBatchSize << std::endl;
+                item->set_layout(ov::Layout(layout.to_string().insert(1, "N,")));
+            }
+            // Get partial shape and layout again (in case it's changed after setting batch dimension)
+            auto shape = item->get_partial_shape();
             layout = item->get_layout();
+            // Set shape at batch index to match overrideModelBatchSize
+            shape[ov::layout::batch_idx(layout)] = overrideModelBatchSize;
+            shapeString = shape.to_string();
         }
-        if (shape[ov::layout::batch_idx(layout)].is_dynamic()) {
-            throw std::logic_error("ERROR: Shape \"" + shape.to_string() + "\"" +
-                                   " has dynamic batch size which is not supported by NPU\n"
-                                   "Cannot apply fixed batch: " +
-                                   std::to_string(batch) +
-                                   ". Please remove the parameter from config: \"override_model_batch_size\"");
-        }
-        ov::set_batch(model, batch);
     }
-}
 
-void reshape(ov::OutputVector inputsInfo, InputsInfo& infoMap, std::shared_ptr<ov::Model>& model,
-             std::string& shapeString, int overrideModelBatchSize, std::string_view device) {
-    std::vector<InputsInfo> infoMaps;
     if (!shapeString.empty()) {
```

Comment:
nitpick:
Dou you think you could add a new comment like: either process an externally requested "shape" or internally reconstructed shape based on override_batch_model request? The comment above it good, but when a reader have reached out this condition, they could forgot what was all about
Or better, I'd suggest we add another function like `std::string prepareModelLayoutForBatchingIfRequested(override_model_batch_size)` which will do the same job you described in you bullet-list comment

By thew way, IMHO I like this new `reshape` as you reused existing routine in adopting `override_model_batch`

Reply:
Yup I'll add more comments to make it clearer how it works within the `reshape` method 👍